agent/requestor/src/payment.rs

```diff
@@ -105,6 +109,8 @@ async fn process_invoice(
     payment_api: PaymentRequestorApi,
     invoice_id: String,
     agreement_allocation: Arc<Mutex<HashMap<String, String>>>,
+    one_agreement: bool,
+    ctrl_c_abort: futures::future::AbortHandle,
```

Comment:
you can pass Option here, and if it is `Some(ctrl_c_abort)` than call abort on it. It will remove one extra param